Cpp_hands_on_programming/test/test.cc

```diff
@@ -0,0 +1,133 @@
+#include <iostream>
+#include <string>
+#include <vector>
+#include "lib/other/Prog_init.h"
+#include "lib/pipe/Ipc_pipe.h"
+#include "lib/queue/Ipc_queue.h"
+#include "lib/shm/Ipc_shm.h"
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+#include <fstream>
+#include <fcntl.h>
+#include "lib/abstract/Ipc_method.h"
+
+
+class Prog_initTest : public::testing::Test {
+protected:
+    Prog_init init;
+    
+    static void SetUpTestSuite() {
+        std::cout << "START PROG_INIT TESTS" << std::endl;
+    }
+
+    static void TearDownTestSuite() {
+        std::cout << "END OF PROG_INIT TESTS" << std::endl;
+    }
+
+    // Prog_initTest() {
+    //     std::cout << "Constructor is called" << std::endl;
+    // }
+
+    // void SetUp() override {
+    //     std::cout << "SetUp is called" << std::endl;
+    // }
+
+    // void TearDown() override {
+    //     std::cout << "TearDown is called" << std::endl;
+    // }
+
+    // ~Prog_initTest() {
+    //     std::cout << "Destructor is called" << std::endl;
+    // }
+
+};
+
+std::map<std::string, methods> input_label = {
+    {"--pipe", PIPES},
+    {"--queue", QUEUE},
+    {"--shm", SHM}
+};
+
+// Parameterized test with class fixture
+class Prog_initParamTest : public Prog_initTest, public ::testing::WithParamInterface<std::pair<const std::string, methods>> {};
+
+TEST_F(Prog_initTest, PrintInstruction) {
+    std::string expected = "This program is a file data transfer program in IPC environment, it is for IPC methods practice.\n";
+    expected += "The program shall receive an argument starts with \"--\"\n";
+    expected += "The options below are available for different file transfer usages(may require argument), --file option is mandatory:\n\n";
+    expected += "--help(-h): display all commands instruction and description(no arguments)\n";
+    expected += "--queue(-q): uses queue as IPC method(argument required)\n";
+    expected += "--pipe(-p): uses name pipe(FIFO) as IPC method(argument required)\n";
+    expected += "--shm(-s): uses shared memory buffer as IPC method(argument required)\n";
+	expected += "--file(-f): file used for read/write data(argument required)\n\n";
+    testing::internal::CaptureStdout();
```

Comment:
As i mentionned in teamchat internal::CaptureStdout() should be avoided.

Reply:
Not really, for other things, there is documentation of Gtest upgrades. This can change without any information because it shall be used internally.